geonode/resource/manager.py

```diff
@@ -78,24 +105,116 @@ def search(self, filter: dict, /, type: object = None) -> QuerySet:
-            return instance.delete()
+        _resource = instance or ResourceManager._get_instance(uuid)
+        if _resource:
+            try:
+                self._resource_manager.delete(uuid, instance=_resource)
+                _resource.get_real_instance().delete()
+                return 1
+            except Exception as e:
+                logger.exception(e)
         return 0
 
-    def create(self, uuid: str, /, instance: ResourceBase = None) -> int:
-        pass
+    @transaction.atomic
+    def create(self, uuid: str, /, type: object = None, defaults: dict = {}) -> ResourceBase:
+        if type.objects.filter(uuid=uuid).exists():
+            raise ValidationError(f'Object of type {type} with uuid [{uuid}] already exists.')
+        _resource, _created = type.objects.get_or_create(
+            uuid=uuid,
+            defaults=defaults)
+        if _resource and _created:
+            try:
+                _resource.set_processing_state(enumerations.STATE_RUNNING)
+                _resource.set_missing_info()
+                _resource = self._resource_manager.create(uuid, type=type, defaults=defaults)
+                _resource.set_processing_state(enumerations.STATE_PROCESSED)
+            except Exception as e:
+                _resource.delete()
+                raise e
+            resourcebase_post_save(_resource)
+        return _resource
 
-    def update(self, uuid: str, /, instance: ResourceBase = None) -> int:
-        pass
+    @transaction.atomic
+    def update(self, uuid: str, /, instance: ResourceBase = None, xml_file: str = None, metadata_uploaded: bool = False,
+               vals: dict = {}, regions: dict = {}, keywords: dict = {}, custom: dict = {}, notify: bool = True) -> ResourceBase:
+        _resource = instance or ResourceManager._get_instance(uuid)
+        if _resource:
+            _resource.set_processing_state(enumerations.STATE_RUNNING)
+            _resource.set_missing_info()
+            _resource.metadata_uploaded = metadata_uploaded
+            logger.debug(f'Look for xml and finalize Layer metadata {_resource}')
+            try:
+                if metadata_uploaded and xml_file:
+                    _md_file = None
+                    try:
+                        _md_file = storage_manager.open(xml_file)
+                    except Exception as e:
+                        logger.exception(e)
+                        _md_file = open(xml_file)
+                    _uuid, vals, regions, keywords, custom = parse_metadata(_md_file.read())
+                    if uuid and uuid != _uuid:
+                        raise ValidationError("The UUID identifier from the XML Metadata is different from the {_resource} one.")
+                    else:
+                        uuid = _uuid
+                logger.debug(f'Update Layer with information coming from XML File if available {_resource}')
+                _resource = metadata_storers(_resource.get_real_instance(), custom)
```

Comment:
Metadata storers, by definition, should be always the last step. Plus on creation are missing

Reply:
I don't wont to store additional metadata at creation time. Ok about the first observation.